lib/AST/Evaluator.cpp

```diff
@@ -378,24 +379,93 @@ void Evaluator::dumpDependenciesGraphviz() const {
   printDependenciesGraphviz(llvm::dbgs());
 }
 
+void evaluator::DependencyCollector::realizeOrRecord(const Reference &ref) {
```

Comment:
It would be cleaner if the evaluator didn't hold a ReferenceNameTracker at all; instead, TypeCheckSourceFileRequest should grab the list of References that were recorded at the end, and pass them to the ReferenceNameTracker right there.